include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -302,7 +302,6 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
 
                 BOOST_ASSERT(weight_vector.size() == id_vector.size());
                 // ed.distance should be total_weight + penalties (turn, stop, etc)
-                BOOST_ASSERT(ed.distance >= total_weight);
```

Comment:
If this is not true anymore, either the comment has to go as well or we need an adjustment of both the comment and the assertion